servers/curated-corpus-api/src/database/types.ts

```diff
@@ -174,6 +174,7 @@ export type ApprovedItem = ApprovedItemModel & {
 
 export type SectionItem = SectionItemModel & {
   approvedItem: ApprovedItem;
+  section?: Pick<SectionModel, 'externalId'>;
```

Comment:
I don't see any code to load this data, so I don't think `section` is ever populated with a `SectionModel`.
```suggestion
  section?: SectionModel;
```
I think it's reasonable to load the Section whenever creating or removing a SectionItem. Where SectionItems are loaded, you would do:
```ts
include: {
  approvedItem: { ... },
  section: true,
}
```